src/Core/WhenExpr/WhenExpr.re

```diff
@@ -0,0 +1,156 @@
+module Value = {
+  [@deriving show({with_path: false})]
+  type t =
+    | String(string)
+    | True
+    | False;
+
+  // Emulate JavaScript semantics
+  let asBool =
+    fun
+    | True => true
+    | False => false
+    | String("") => false
+    | String(_) => true;
+
+  // Emulate JavaScript semantics
+  let asString =
+    fun
+    | True => "true"
+    | False => "false"
+    | String(str) => str;
+};
+
+[@deriving show({with_path: false})]
+type t =
+  | Defined(string)
+  | Eq(string, Value.t)
+  | Neq(string, Value.t)
+  | Regex(string, option(Re.re))
```

Comment:
Wow, you brought in Regex compatability too - nice!

I wonder how well the `Re` regular expression map to use the regular expression syntax used in VSCode. If we find that we aren't supporting some regular expression constructs that VSCode uses, we could see if Oniguruma handles those. No action needed; just something to think about.

Reply:
Yeah, I couldn't find a good source for what "standards" oniguruma tries to support, but they're likely also mutually exclusive to some extent, so that vagueness didn't fill me with confidence either. Hopefully the regexes used just aren't all that advanced.